File: frontend/scripts/domain/ChatRoomMessage.ts
export class ChatRoomMessage {

  static fromPOJO(params: Object): ChatRoomMessage {
    let chatRoomMessage: ChatRoomMessage = new ChatRoomMessage();

    chatRoomMessage.body = params['body'];
    chatRoomMessage.timestamp = new Date(params['timestamp']);
    chatRoomMessage.senderName = params['sender']['name'];

    return chatRoomMessage;
  }

  private body: string;
  private senderName: string;
  private timestamp: Date;

  public getBody(): string {
    return this.body;
  }

  public getTimestamp(): Date {
    return this.timestamp;
  }

  public getSenderName(): string {
    return this.senderName;
  }

}

File: src/main/java/l2k/trivia/server/domain/chat/ChatRoomMessage.java
package l2k.trivia.server.domain.chat;

import java.util.Date;

import l2k.trivia.server.domain.User;

public class ChatRoomMessage {
	
	protected String body;
	protected Sender sender;
	protected Date timestamp;
	
	public ChatRoomMessage() {}
	
	public ChatRoomMessage(User user, String body) {
		setSender(user);
		setBody(body);
		setTimestamp(new Date());
	}

	public String getBody() {
		return body;
	}

	public void setBody(String body) {
		this.body = body;
	}

	public Sender getSender() {
		return sender;
	}

	public void setSender(Sender sender) {
		this.sender = sender;
	}

	public Date getTimestamp() {
		return timestamp;
	}

	public void setTimestamp(Date timestamp) {
		this.timestamp = timestamp;
	}

}
